lib/core/src/Cardano/Wallet/DB/Store/Wallets/Store.hs

```diff
@@ -57,68 +68,87 @@ import qualified Cardano.Wallet.DB.Store.Transactions.Model as TxStore
 import qualified Cardano.Wallet.Primitive.Types as W
 import qualified Data.Map.Strict as Map
 
+mkStoreMetaWithSubmissions :: W.WalletId
+    -> Store (SqlPersistT IO) DeltaWalletsMetaWithSubmissions
+mkStoreMetaWithSubmissions wid =
+    embedStore' embedConstrainedSubmissions
+    $ pairStores
+        (mkStoreMetaTransactions wid)
+        (mkStoreSubmissions wid)
+
 -- | Store for 'WalletsMeta' of multiple different wallets.
 mkStoreWalletsMeta :: Store
         (SqlPersistT IO)
-        (DeltaMap W.WalletId DeltaTxMetaHistory)
+        (DeltaMap W.WalletId DeltaWalletsMetaWithSubmissions)
 mkStoreWalletsMeta =
     Store
     { loadS = load
     , writeS = write
     , updateS = update
     }
   where
-    write reset = forM_ (Map.assocs reset) $ \(wid,metas)
-        -> writeS (mkStoreMetaTransactions wid) metas
-    update _ (Insert wid metas) =
+    write reset = forM_ (Map.assocs reset) $ \(wid,(metas,subs)) -> do
+        writeS (mkStoreMetaTransactions wid) metas
+        writeS (mkStoreSubmissions wid) subs
+    update :: Map W.WalletId (TxMetaHistory, TxLocalSubmissionHistory)
+        -> DeltaMap W.WalletId DeltaWalletsMetaWithSubmissions
+        -> SqlPersistT IO ()
+    update _ (Insert wid (metas,subs)) = do
         writeS (mkStoreMetaTransactions wid) metas
+        writeS (mkStoreSubmissions wid) subs
     update _ (Delete wid) = do
         deleteWhere [TxMetaWalletId ==. wid ]
-    update _ (Adjust wid da) =
+        deleteWhere [LocalTxSubmissionWalletId ==. wid ]
+    update _ (Adjust wid (ChangeMeta da)) = do
         updateS (mkStoreMetaTransactions wid) undefined da
-    load :: SqlPersistT
-            IO
-            (Either SomeException (Map.Map W.WalletId TxMetaHistory))
+    update _ (Adjust wid (ChangeSubmissions da)) = do
+        updateS (mkStoreSubmissions wid) undefined da
```

Comment:
Same comment as for `embedConstrainedSubmissions`: It's not clear to me that this has the same effect on the database as the constraints imposed by `apply` have on the in-memory type. — However, this could be the same due to the delete cascade and foreign key constraint that we have. 🤪

At the very least, I think we should document this point somewhere. — I think the best way to go about that may be to use only `mkStoreMetaWithSubmissions` instead of `mkStoreMetaTransactions` and `mkStoreSubmissions` in the definition of `mkStoreWalletsMeta`, as this would put the "source of truth" in the `mkStoreMetaWithSubmissions` thing (with or without embedding), where we can document it.